### DISCUSSION TITLE
Migrating GitHub Classroom Assignment Repository Creation from “Create from Template” to Forks [Public Beta] problem

### DISCUSSION THREAD

**Discussion by jteno** [community/community#106732] [GitHub Education]
Hi,

I've successfully setup classroom/assignments/autograding via a template repository (The old model). I noticed that this model is going to be migrated to forks in the summer. I wanted to try the fork model and I've opted-in to utilise the forking in a new classroom, and assignment I've created to test how this is supposed to work

I've enabled private forking in my organisation and, I've created a new private repository that I've setup as a new repository / template. After this I created a new classroom and published a new assignment that uses the new template repository as starter code. When I send the invitation to a student, they are able to accept the assignment, but after this they get an error:

1. "Repository Access Issue
You no longer have access to your assignment repository. Contact your teacher for support"

If I try to access the repository link via GitHub Classroom I get an 2: 

"500 -Internal server error
We're sorry, but something went wrong.
[Try going back to classroom dashboard](https://classroom.github.com/)"

Logging in with the student account to GitHub.com, I can see and access the repository from the organization, but all GitHub Actions and Workflows are disabled by default. 

What am I missing?

**Comment by naish-uwo** [comment]
This also happening for classrooms that I've set up in two different organizations. The student repository is forked from the template, but the links within the classroom are broken for the student and me. The generated repo can be accessed directly from GitHub.

Everything was working properly up until sometime last evening/early morning today.

**Comment by toksaitov** [comment]
We are experiencing the same issue, which began on Monday. When a student visits the link to create a repository, the student's repository is initially created correctly (as manually verified by observing the list of repositories on github.com). However, on the GitHub Classroom invitation page, where we expect the link to the new repository to be displayed, a 'Repository Access Issue' message appears upon refreshing. We are using the new forked repository workflow in beta.

**Comment by acbart** [comment]
Yep, same error for me today with one of my repositories. This will be a huge problem if it's happening outside of my account. Haven't heard anything from students yet, but I hope this is being handled!

**Reply by acbart** [reply]
The workaround has been that the repo does get created during this process. If they go to the organization's repos, they can see it.

**Reply by jteno** [reply]
How is this a workaround if the actions, autograder etc are empty?

**Comment by dendy-prtha-tiukdw** [comment]
Yeah, the issue still happened now, when i create new assignment using private repo as template, and set repository visiblity as public. It seems it try to apply "Student repository visibility will be inherited from the starter code repository." rules, but the UI still have option to select repository visibility. This problem was not happened last week.
Below is my classroom that not have repository visibility when create assignment
![image](https://github.com/community/community/assets/90243581/b0af9c29-6bb2-4fc3-88b0-d57cbe124170)

Here my another classroom that have repository visibility when create assignment
![image](https://github.com/community/community/assets/90243581/91c10311-b8dd-482e-a2d3-897ea17276c1)

**Comment by AminaEmenena** [comment]
I am having the same issue

`Repository Access Issue
You no longer have access to your assignment repository. Contact your teacher for support`

**Comment by MisterPl0w** [comment]
I'm having the same issue - was going to start with my class tomorrow and had a student try it out for me. Failure after failure. Glad I found this discussion. I guess I'll go back to the non-fork method for now? I'm _**very**_ new to Github Classroom.

**Comment by jteno** [comment]
I've reported this as a bug to GitHub.

**Comment by FelixBenning** [comment]
I have the same issue, the repository seems to have been created, a team was created with access to this repository, but the new repository does not contain github actions and it thinks there is an access isssue on the invitation page.

I also have the server error 500 error code when I try to access it from the classroom dashboard. 

Could we get some feedback from the GitHub Team for the timeline on this? I would like to know whether or not I have to redo the assignment in a template friendly way (I created a solution branch since only the main branch is forked, but this is not the case for the template mechanism)

**Comment by jteno** [comment]
Anyone from Github.com willing to comment on this? I think this is a big issue if you are forcing us to update to forked repos and this isn't working. I haven't received any reply from my bug report at all.

**Comment by jteno** [comment]
Anyone tried this again? I submitted a bug report to GitHub.com 13.2 and haven't received any reply to this. Rather odd support.

**Comment by parsa-rajabi-ubc** [comment]
Hi folks, does anyone have an update on this? This issue seems to have bubbled up again. ([Link to thread on this](https://github.com/orgs/community/discussions/138928))

**Reply by parsa-rajabi-ubc** [reply]
@github-staff

**Comment by jteno** [comment]
I just tested this by creating a new classroom, a template repository for starter code, and auto grading scripts for assignments.  

Everything works from creating the template repository in our organization, and utilising it as a started code for the students and creating a new classroom and assignments with the starter code. 

I also tested the update of the assignment and pull requests from the assignment to students, this works also. I'm going to move forward with this implementation in our git course next spring, hopefully nothing gets changed.

**Comment by JeffreyWalmer** [comment]
Did anyone here could solve this? I am having this same issue

**Comment by jacksonlewis97** [comment]
Still getting this issue. Never had a problem for 2 years and now this is happening. Any fix? It is just a few of my students, not all of them.

**Reply by astump** [reply]
Same for me!  Only seems to affect some students.

**Comment by nikmomo** [comment]
I've developed a service that resolves this issue efficiently by sending a new invitation. It can be deployed on a server for responsive operation.
https://github.com/nikmomo/GHClassroom.AccessFix

**Comment by ssyberg** [comment]
Just noting my class this year is experiencing this a ton as well during every assignment/lab, it's quite disruptive. Never happened in previous years for us.
